x-pack/test/alerting_api_integration/security_and_spaces/tests/alerting/alerts.ts

```diff
@@ -603,6 +603,140 @@ export default function alertTests({ getService }: FtrProviderContext) {
               throw new Error(`Scenario untested: ${JSON.stringify(scenario)}`);
           }
         });
+
+        it(`shouldn't schedule actions when alert is muted`, async () => {
+          const reference = `create-test-8:${user.username}`;
+          const createdAction = await createIndexRecordAction(space.id);
+          const response = await supertestWithoutAuth
+            .post(`${getUrlPrefix(space.id)}/api/alert`)
+            .set('kbn-xsrf', 'foo')
+            .auth(user.username, user.password)
+            .send(
+              getTestAlertData({
+                enabled: false,
+                interval: '1s',
+                alertTypeId: 'test.always-firing',
+                alertTypeParams: {
+                  index: ES_TEST_INDEX_NAME,
+                  reference,
+                },
+                actions: [
+                  {
+                    group: 'default',
+                    id: createdAction.id,
+                    params: {
+                      index: ES_TEST_INDEX_NAME,
+                      reference,
+                      message: 'from:default',
+                    },
+                  },
+                ],
+              })
+            );
+
+          switch (scenario.id) {
+            case 'no_kibana_privileges at space1':
+            case 'space_1_all at space2':
+            case 'global_read at space1':
+              expect(response.statusCode).to.eql(404);
+              expect(response.body).to.eql({
+                statusCode: 404,
+                error: 'Not Found',
+                message: 'Not Found',
+              });
+              break;
+            case 'space_1_all at space1':
+            case 'superuser at space1':
+              await supertestWithoutAuth
+                .post(`${getUrlPrefix(space.id)}/api/alert/${response.body.id}/_mute`)
+                .set('kbn-xsrf', 'foo')
+                .auth(user.username, user.password)
+                .expect(204, '');
+              await supertestWithoutAuth
+                .post(`${getUrlPrefix(space.id)}/api/alert/${response.body.id}/_enable`)
+                .set('kbn-xsrf', 'foo')
+                .auth(user.username, user.password)
+                .expect(204, '');
+              // Wait until alerts scheduled actions twice to ensure actions had a chance to execute once
+              await esTestIndexTool.waitForDocs('alert:test.always-firing', reference, 2);
+              const executedActionsResult = await esTestIndexTool.search(
+                'action:test.index-record',
+                reference
+              );
+              expect(executedActionsResult.hits.total.value).to.eql(0);
+              break;
+            default:
+              throw new Error(`Scenario untested: ${JSON.stringify(scenario)}`);
+          }
+        });
+
+        it(`shouldn't schedule actions when alert instance is muted`, async () => {
+          const reference = `create-test-9:${user.username}`;
```

Comment:
Seems like all this duplicated code could be extracted into a function.  The only difference here is the muting of an instance versus the muting of an alert.

Reply:
Changes / mini refactor has been made in the following three commits: https://github.com/elastic/kibana/pull/43712/commits/413e67aa3bcbda1baeefbdbaaf8422e7b108a9c4, https://github.com/elastic/kibana/pull/43712/commits/491887f15cf3d636467b9b54b9196ce881c9240b and https://github.com/elastic/kibana/pull/43712/commits/8d00147d65c659a2fc10821d2d90153760a60185.